src/box/lua/net_replicaset.lua

```diff
@@ -0,0 +1,376 @@
+local fiber = require('fiber')
+local net_box = require('net.box')
+
+-- The constant is copy-paste from tarantool/src/lua/socket.lua.
+local TIMEOUT_INFINITY = 500 * 365 * 86400
+
+-- Error type used for all errors raised in net.replicaset.
+local ERROR_TYPE = 'NetReplicasetError'
+
+-- Map of errors by their codes.
+local error_list = {
+    [1] = {
+        name = 'REPLICASET_NOT_FOUND',
+        message = 'The replicaset was not found by its name',
+    },
+    [2] = {
+        name = 'NO_WRITABLE',
+        message = 'Writable instance was not found in replicaset',
+    },
+    [3] = {
+        name = 'MORE_THAN_ONE_WRITABLE',
+        message = 'More than one writable was found in replicaset',
+    },
+}
+
+-- Map of error codes by error name.
+local error_codes = {}
+for code, err in ipairs(error_list) do
+    error_codes[err.name] = code
+end
+
+-- Raise a net.replicaset specific error.
+local function rs_error(err_code, replicaset_name, payload)
```

Comment:
Now we patched box.error (added `level` option), so please review the code to provide the functional (add argument `level`, and set `level` to default value (if a user doesn't use the argument), to point error place)